src/main/java/com/ldtteam/blockout/views/TabSet.java

```diff
@@ -0,0 +1,271 @@
+                int cap = Integer.parseInt(m.group(1));
+                tabIcons = new ResourceLocation[cap];
+                for (int i = 1 ; i <= cap; i++)
+                {
+                    tabIcons[i-1] = new ResourceLocation(
+                      res.getNamespace(),
+                      res.getPath().replace(m.group(0), String.valueOf(i)));
+                }
+            }
+            else
+            {
+                tabIcons = new ResourceLocation[] {res};
+            }
+        });
+
+        tabs = new View();
+        tabs.setID("tabs");
+        addChild(0, tabs);
+        if (vertical)
+        {
+            tabs.setPosition(overlap - tabWidth, offset);
+            tabs.setSize(tabWidth, height - 2*offset);
+        }
+        else
+        {
+            tabs.setPosition(offset, overlap - tabHeight);
+            tabs.setSize(width - 2*offset, tabHeight);
+        }
+
+        spacing = params.getInteger("spacing", 2);
+    }
+
+    /**
+     * Selects a tab from its index in the tab button list
+     * @param index the index to select
+     */
+    public void setSelectedTab(int index)
+    {
+        if (index >= tabs.children.size()) return;
+        Pane tab = tabs.children.get(index);
+        if (!(tab instanceof ButtonTab)) return;
+        selectedTab = ((ButtonTab) tab).toSelect;
+
+        children.stream()
+          .filter(child -> child instanceof Tab)
+          .forEach(child -> child.setVisible(child.equals(selectedTab)));
+    }
+
+    /**
+     * Changes the selected tab view to the given one
+     * @param tab the tab to switch to
+     */
+    public void setSelectedTab(Tab tab)
+    {
+        children.stream()
+          .filter(child -> child instanceof Tab)
+          .forEach(child -> {
+              child.setVisible(child.equals(tab));
+              if (child.isVisible()) selectedTab = tab;
+          });
+    }
+
+    /**
+     * Adds the given tab to this tab set
+     * @param tab the tab to add
+     */
+    public void addTab(Tab tab)
+    {
+        tab.padding = childPadding;
+        addChild(tab);
+        new ButtonTab(tab);
+    }
+
+    /**
+     * Adds the given tab to this tab set
+     * @param tab the tab to add
+     * @param button an existing ButtonTab to link to the given tab
+     */
+    public void addTab(Tab tab, ButtonTab button)
+    {
+        tab.padding = childPadding;
+        button.toSelect = tab;
+        addChild(tab);
+    }
+
+    @Override
+    public void parseChildren(@NotNull final PaneParams params)
+    {
+        super.parseChildren(params);
+
+        getChildren().stream()
+          .filter(child -> child instanceof Tab)
+          .map(child -> (Tab) child)
+          .peek(child -> child.padding = childPadding)
+          .forEach(ButtonTab::new);
+
+        setSelectedTab(0);
+    }
+
+    @Override
+    protected boolean childIsVisible(final Pane child)
+    {
+        // Draw the tabs despite being potentially out of the tab set view
```

Comment:
This should be elemently-inlined (not sure how to say properly :D ) just like scrollbar in ScrollingView - this overall hard to design.
We should propably invent a better mechanism for auto elements like this and scrollbar: something like the element is in the xml file with fixed id which allows the xml writer to define it's size and other arguments - this obviously add unwanted xml which you have to remember to put in.
Feel free to come with better solution